numba/core/interpreter.py

```diff
@@ -314,6 +314,164 @@ def peep_hole_delete_with_exit(func_ir):
     return func_ir
 
 
+def peep_hole_fuse_dict_add_updates(func_ir):
+    """
+    This rewrite removes d1.update(d2) calls that
+    are between two dictionaries, d1 and d2 in the
+    same basic block, resulting from a Python 3.10
+    upgrade. If both are constant dictionaries
+    defined in that block and neither is used between
+    the update call, then we replace d1 with a new definition
+    that combines the two dicitonaries.
+
+    Python 3.10 may also rewrite a dictionary as an empty
+    build_map + many map_add, so we also need to replace those
+    expressions with a constant build map.
+    """
+
+    def build_new_buildmap(name, blk, old_lineno, new_items):
+        """
+            Create a new buildmap with a new set of items
+            but all the other info the same.
+        """
+        old_assign = blk.body[old_lineno]
+        old_target = old_assign.target
+        old_bm = old_assign.value
+        # Build the literals
+        literal_keys = []
+        # Track the values
+        values = []
+        for pair in new_items:
+            k, v = pair
```

Comment:
why not `for k, v in new_items`?

Reply:
This was just from debugging. Will update.